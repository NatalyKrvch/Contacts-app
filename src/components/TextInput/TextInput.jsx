import { Form } from 'react-bootstrap'

const TextInput = ({
  label,
  errorMessage,
  value,
  onChange,
  className,
  id,
  type = 'text',
  required = false,
}) => {
  return (
    <Form.Group controlId={id}>
      <Form.Label className={className}>{label}</Form.Label>
      <Form.Control
        type={type}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        isInvalid={!!errorMessage}
        required={required}
      />
      <Form.Control.Feedback type="invalid">
        {errorMessage}
      </Form.Control.Feedback>
    </Form.Group>
  )
}

export default TextInput
