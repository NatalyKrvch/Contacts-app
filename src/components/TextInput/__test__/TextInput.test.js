import { render, screen, fireEvent } from '@testing-library/react';
import TextInput from '../TextInput';

describe('TextInput', () => {
  const mockProps = {
    label: 'Test Label',
    errorMessage: '',
    value: '',
    onChange: jest.fn(),
    className: 'test-class',
    id: 'test-input',
    type: 'text',
    placeholder: 'Enter text',
    required: false,
  };

  it('renders component correctly', () => {
    const { container } = render(<TextInput {...mockProps} />);
    
    expect(container).toMatchSnapshot();
  });

  it('renders input with label', () => {
    const { container } = render(<TextInput {...mockProps} />);
    
    expect(screen.getByLabelText('Test Label')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Enter text')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('calls onChange when input value changes', () => {
    const { container } = render(<TextInput {...mockProps} />);
    
    fireEvent.change(screen.getByLabelText('Test Label'), { target: { value: 'new value' } });
    expect(mockProps.onChange).toHaveBeenCalledWith('new value');
    expect(container).toMatchSnapshot();
  });

  it('displays error message when errorMessage is provided', () => {
    const propsWithError = { ...mockProps, errorMessage: 'Error occurred' };
    const { container } = render(<TextInput {...propsWithError} />);
    
    expect(screen.getByText('Error occurred')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders required attribute when required is true', () => {
    const requiredProps = { ...mockProps, required: true };
    const { container } = render(<TextInput {...requiredProps} />);
    
    expect(screen.getByLabelText('Test Label')).toBeRequired();
    expect(container).toMatchSnapshot();
  });
});

