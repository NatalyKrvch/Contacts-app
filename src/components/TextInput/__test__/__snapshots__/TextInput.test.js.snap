// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput calls onChange when input value changes 1`] = `
<div>
  <div>
    <label
      class="test-class form-label"
      for="test-input"
    >
      Test Label
    </label>
    <input
      class="form-control"
      id="test-input"
      placeholder="Enter text"
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</div>
`;

exports[`TextInput displays error message when errorMessage is provided 1`] = `
<div>
  <div>
    <label
      class="test-class form-label"
      for="test-input"
    >
      Test Label
    </label>
    <input
      class="form-control is-invalid"
      id="test-input"
      placeholder="Enter text"
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    >
      Error occurred
    </div>
  </div>
</div>
`;

exports[`TextInput renders component correctly 1`] = `
<div>
  <div>
    <label
      class="test-class form-label"
      for="test-input"
    >
      Test Label
    </label>
    <input
      class="form-control"
      id="test-input"
      placeholder="Enter text"
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</div>
`;

exports[`TextInput renders input with label 1`] = `
<div>
  <div>
    <label
      class="test-class form-label"
      for="test-input"
    >
      Test Label
    </label>
    <input
      class="form-control"
      id="test-input"
      placeholder="Enter text"
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</div>
`;

exports[`TextInput renders required attribute when required is true 1`] = `
<div>
  <div>
    <label
      class="test-class form-label"
      for="test-input"
    >
      Test Label
    </label>
    <input
      class="form-control"
      id="test-input"
      placeholder="Enter text"
      required=""
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</div>
`;
